hipogram/posts/views.py

```diff
@@ -39,40 +47,29 @@ def post_new(request):
         form = PostForm()
     return render(request, 'share.html', {'form': form})
     
-#to delete a post
-def delete_post(request, pk):
-    template_name = 'update.html'
-    post = get_object_or_404(Post, pk=pk)
-
-    try:
-        if request.method == 'POST':
-            form = PostForm(request.POST, instance=post)
-            post.delete()
-            messages.success(request, 'You have successfully deleted the post')
-        else:
-            form = PostForm(instance=post)
-    except Exception as e:
-        messages.warning(request, 'The post could not be deleted: Error {}'.format(e))
-
-    context = {'form':form,}
+@login_required()
+def delete_post(request, id):
+    post = get_object_or_404(Post, id=id)
+    if request.method == 'POST':
+        form = PostForm(request.POST, instance=post)
+        post.delete()
+        messages.success(request, 'You have successfully deleted the post')
```

Comment:
Good addition :+1: 

However, I believe it doesn't show up at the current state of the PR.